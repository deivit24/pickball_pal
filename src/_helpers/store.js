import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
