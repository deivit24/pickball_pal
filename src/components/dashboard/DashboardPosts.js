/* eslint-disable no-undef */
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { getUserPosts } from '../../_actions/post';
import Loading from '../layout/Loading';
import PostCard from '../posts/PostCard';

import '../../assets/css/Dashboard.css';

const DashboardPosts = () => {
  const post = useSelector((store) => store.post);
  const dispatch = useDispatch();

  const { posts, loading } = post;
  useEffect(() => {
    dispatch(getUserPosts());
  }, [dispatch]);

  let result;

  if (loading) {
    result = <Loading type="spin" color="#3f50b5" />;
  }
  if (posts.length === 0) {
    result = (
      <>
        <div id="DashboardPosts" className="container mt-5">
          <h4>You haven't posted anything yet</h4>
        </div>
      </>
    );
  } else {
    result = (
      <>
        <div id="DashboardPosts" className="container mt-5">
          {posts.map((post) => (
            <PostCard key={post._id} post={post} />
          ))}
        </div>
      </>
    );
  }

  return <>{result}</>;
};

export default DashboardPosts;
