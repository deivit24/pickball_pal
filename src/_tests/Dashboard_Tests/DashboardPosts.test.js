import React from 'react';
import { render } from '@testing-library/react';
import DashboardPosts from '../../components/dashboard/DashboardPosts';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing DashboardPosts

describe('Testing the DashboardPosts', () => {
  it('it renders DashboardPosts without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <DashboardPosts />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <DashboardPosts />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
