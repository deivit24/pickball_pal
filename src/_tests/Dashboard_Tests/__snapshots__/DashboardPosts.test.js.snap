// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the DashboardPosts matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-5"
    id="DashboardPosts"
  >
    <h4>
      You haven't posted anything yet
    </h4>
  </div>
</DocumentFragment>
`;
